mux.go

```diff
@@ -13,6 +13,10 @@ import (
 	"strings"
 )
 
+var (
+	ErrMethodMismatch = errors.New("Method is Not Allowed")
```

Comment:
Since we don't use the text for this, we should either:

* Lowercase this as convention - e.g. `method is not allowed`
* Use an enum to reflect the type of error it is.

String errors are inflexible and fragile.